shell.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+	let
+	overrides = (builtins.fromTOML (builtins.readFile ./rust-toolchain.toml));
+	libPath = with pkgs; lib.makeLibraryPath [
+		# load external libraries that you need in your rust project here
+	];
+in
+pkgs.mkShell rec {
```

Comment:
```suggestion
pkgs.mkShellNoCC rec {
```

Smaller footprint, if you don't need a C compiler.

Reply:
It's still needed for the c dependencies like OpenSSL and friends isn't it?

Reply:
Maybe, if in doubt just try and error.

Reply:
Rust also needs ld and that isn't in the no-cc shell, right? I recall having problems with something like that I think.

Reply:
Tested it. Works with NoCC so I changed to that